compiler/src/iree/compiler/Codegen/SPIRV/Passes.cpp

```diff
@@ -335,11 +339,13 @@ void addSPIRVMatmulPromoteVectorizePassPipeline(OpPassManager &pm,
   nestedModulePM.addNestedPass<func::FuncOp>(createForOpCanonicalizationPass());
   nestedModulePM.addPass(createCanonicalizerPass());
   nestedModulePM.addPass(createCSEPass());
+  nestedModulePM.addNestedPass<func::FuncOp>(
+      memref::createFoldMemRefAliasOpsPass());
   nestedModulePM.addNestedPass<func::FuncOp>(
       createOptimizeVectorTransferPass());
 
-  nestedModulePM.addNestedPass<func::FuncOp>(
-      createGPUPipeliningPass(pipelineDepth ? pipelineDepth : 1));
+  nestedModulePM.addNestedPass<func::FuncOp>(createGPUPipeliningPass(
+      /*epiloguePeeling=*/true, pipelineDepth, storeStage));
```

Comment:
Might want to run canonicalization and cse pass to clean up stuff a bit after this? Or preferably, do it at the end of `addLoopMaterializationPasses` for all pipelines. (I suspect it will also make `CHECK` matching in tests easier.)